cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -364,23 +383,63 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 
 func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
 	tagValues := getOpenAPITagValue(CommentLines)
-	anyExtension := false
+	type NameValue struct {
+		Name, Value string
+	}
+	extensions := []NameValue{}
 	for _, val := range tagValues {
 		if strings.HasPrefix(val, tagExtensionPrefix) {
-			if !anyExtension {
-				g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
-				anyExtension = true
-			}
 			parts := strings.SplitN(val, ":", 2)
 			if len(parts) != 2 {
 				return fmt.Errorf("Invalid extension value: %v", val)
 			}
-			g.Do("\"$.$\": ", parts[0])
-			g.Do("\"$.$\",\n", parts[1])
+			extensions = append(extensions, NameValue{parts[0], parts[1]})
 		}
 	}
-	if anyExtension {
-		g.Do("},\n},\n", nil)
+	patchMergeKeyTag, err := getSingleTagsValue(CommentLines, tagPatchMergeKey)
+	if err != nil {
+		return err
+	}
+	if len(patchMergeKeyTag) > 0 {
+		extensions = append(extensions, NameValue{patchMergeKeyExtensionKey, patchMergeKeyTag})
+	}
+	patchStrategyTag, err := getSingleTagsValue(CommentLines, tagPatchStrategy)
+	if err != nil {
+		return err
+	}
+	if len(patchStrategyTag) > 0 {
+		extensions = append(extensions, NameValue{patchStrategyExtensionKey, patchStrategyTag})
+	}
+	if len(extensions) == 0 {
+		return nil
+	}
+	g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
+	for _, extension := range extensions {
+		g.Do("\"$.$\": ", extension.Name)
+		g.Do("\"$.$\",\n", extension.Value)
+	}
+	g.Do("},\n},\n", nil)
+	return nil
+}
+
+// TODO(#44005): Move this validation outside of this generator (probably to policy verifier
+func (g openAPITypeWriter) validatePatchTags(m *types.Member, parent *types.Type) error {
+	patchMergeKeyStructTag, patchStrategyStructTag := getPatchTags(m)
+	patchMergeKeyCommentTag, err := getSingleTagsValue(m.CommentLines, tagPatchMergeKey)
+	if err != nil {
+		return err
+	}
+	patchStrategyCommentTag, err := getSingleTagsValue(m.CommentLines, tagPatchStrategy)
+	if err != nil {
+		return err
+	}
+	if patchMergeKeyStructTag != patchMergeKeyCommentTag {
```

Comment:
IMO we want always enforce they match, at least for now.
Otherwise, if the client move to use openapi someday and the server is still using the struct tags and they are inconsistent. Patch will do the wrong things.